querybook/webapp/hooks/useCommand.ts

```diff
@@ -0,0 +1,74 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+import { AICommandType } from 'const/aiAssistant';
+import { IQueryCellCommand, QueryCellCommandType } from 'const/command';
+import { useAISocket } from 'hooks/useAISocket';
+
+const COMMANDS_TO_AI_COMMAND_TYPE = {
+    generate: AICommandType.TEXT_TO_SQL,
+    edit: AICommandType.TEXT_TO_SQL,
+    fix: AICommandType.SQL_FIX,
+};
+
+type SyncCommandRunner = (kwargs: Record<string, any>) => any;
+type AsyncCommandRunner = (
+    kwargs: Record<string, string>,
+    callback: (data: any) => void
+) => void;
+
+export type CommandRunner = SyncCommandRunner | AsyncCommandRunner;
+
+export const useCommand = (
+    command: IQueryCellCommand,
+    commandRunner?: CommandRunner,
+    resetResult?: boolean
+) => {
+    const [isRunning, setIsRunning] = useState(false);
+    const [commandResult, setCommandResult] = useState<any>();
+
+    const socket = useAISocket(command.aiCommand, setCommandResult);
+
+    useEffect(() => {
+        if (resetResult) {
+            setCommandResult(undefined);
+        }
+    }, [resetResult]);
```

Comment:
why is a prop used to control commandResult should be wiped? shouldn't this hook return a resetResult callback?

Reply:
My mind suddenly went blank :(